src/wasm/wasm-ir-builder.cpp

```diff
@@ -1924,49 +1936,101 @@ Result<> IRBuilder::makeContBind(HeapType contTypeBefore,
   return Ok{};
 }
 
-Result<> IRBuilder::makeContNew(HeapType ct) {
-  if (!ct.isContinuation()) {
-    return Err{"expected continuation type"};
-  }
-  ContNew curr;
-  curr.contType = ct;
-  CHECK_ERR(visitContNew(&curr));
+Result<> IRBuilder::makeSuspend(Name tag) {
+  Suspend curr(wasm.allocator);
+  curr.tag = tag;
+  curr.operands.resize(wasm.getTag(tag)->sig.params.size());
+  CHECK_ERR(visitSuspend(&curr));
 
-  push(builder.makeContNew(ct, curr.func));
+  std::vector<Expression*> operands(curr.operands.begin(), curr.operands.end());
+  push(builder.makeSuspend(tag, operands));
   return Ok{};
 }
 
 Result<> IRBuilder::makeResume(HeapType ct,
                                const std::vector<Name>& tags,
-                               const std::vector<Index>& labels) {
+                               const std::vector<Index>& labels,
+                               const std::vector<bool>& onTags) {
   if (!ct.isContinuation()) {
     return Err{"expected continuation type"};
   }
   Resume curr(wasm.allocator);
   curr.contType = ct;
   curr.operands.resize(ct.getContinuation().type.getSignature().params.size());
+  curr.handlerTags.set(tags);
+  curr.onTags.set(onTags);
```

Comment:
These don't affect how children get popped off the stack, so there's no need to set them here. In the loop below we can use `onTags` instead of `curr.onTags`.